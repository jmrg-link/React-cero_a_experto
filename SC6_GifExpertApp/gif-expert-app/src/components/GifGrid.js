import React from 'react'
import config from '../../../../config/index.mjs'

export const GifGrid = ({category}) => {
    
    const getGifts = async() => {
        const url = `https://api.giphy.com/v1/gifs/search?q=morty&limit=10&api_key=${config.apikey}`
        const resp = fetch(url)
        const data = await resp.json()
        console.log(data);

    }
  return (
    <>
        <h3>{ category }</h3>
    </>
  )
}
