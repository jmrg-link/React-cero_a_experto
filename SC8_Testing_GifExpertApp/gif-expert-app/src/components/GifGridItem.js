import React from 'react'

export const GifGridItem = ({ id , title , url }) => {
  return (
    <div className="card animate__animated animate__lightSpeedInRight 2s">
        <img src={ url } alt={ title } />
        <p> { title } </p>
    </div>
  )
}
