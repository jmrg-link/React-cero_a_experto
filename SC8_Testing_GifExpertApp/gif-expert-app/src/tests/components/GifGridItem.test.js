import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas en <GifGridItem />", () => {
    const title = 'Un titulo'
    const url   = 'https://localhost/ejemplo.jpeg'
  test("debe de mostrar el componente correctamente", () => {
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
