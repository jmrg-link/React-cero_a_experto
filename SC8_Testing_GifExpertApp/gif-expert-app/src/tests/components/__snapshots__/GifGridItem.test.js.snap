// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__lightSpeedInRight 2s"
>
  <img />
  <p>
     
     
  </p>
</div>
`;
